server/hwapi/endpoints/certification/certification.py

```diff
@@ -61,24 +61,40 @@ def check_certification(
     vendor = repository.get_vendor_by_name(db, system_info.vendor)
     if not vendor:
         return NotCertifiedResponse()
+
     # Match against board and bios
     try:
-        board, bios = logic.find_main_hardware_components(
+        board, bios_list = logic.find_main_hardware_components(
             db, system_info.board, system_info.bios
         )
         related_machine = logic.find_certified_machine(
-            db, system_info.architecture, board, bios
+            db, system_info.architecture, board, bios_list
         )
     except ValueError:
+        logging.error(
+            (
+                "Hardware cannot be found. Machine vendor: %s, model: %s"
+                ", board model: %s, board version: %s, bios version: %s"
+            ),
+            system_info.vendor,
+            system_info.model,
+            system_info.board.product_name,
+            system_info.board.version,
+            system_info.bios.version if system_info.bios else None,
+        )
```

Comment:
Matias specifically asked for logging hardware mismatch data since we'll probably have to adjust the logic according to, for instance, how many certified machines run a newer bios version, different from which we used for issuing a certificate

Reply:
This sounds like we need to store this data to be able to query it rather than just log it. I'm not sure if simply logging will help us get such information. @mz2 wdyt?

Reply:
Fair point, it probably is better that way. I was imagining deriving metrics from logs, but explicitly storing the full data is probably a better idea.

Reply:
After discussing it with @omar-selo, we've decided to add this change in another PR and create a table for storing hardware mismatches